### PR TITLE
Fix compilation warnings for elixir ~> 1.14

### PR DESCRIPTION
**Pull request by juanm-roa**
* Fix compilation warnings
* Update readme

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,11 +23,15 @@ It is currently a standalone package, but may be integrated into Mix at some poi
 
 Version `2.1.1` or lower:
 
-* Elixir ~> `1.8.x` or lower.
+- Elixir ~> `1.8.x` or lower.
 
 Version `3.0.0` or higher:
```
Comment (CristhianRodriguezMolina) [inline README.md]: I think that this should be replaced by ```Version `3.0.0`:```

File: lib/distillery/releases/assembler.ex [hunk 2]
```diff
@@ -655,7 +655,7 @@ defmodule Distillery.Releases.Assembler do
                {:ok, tokens, _} <- :erl_scan.string(String.to_charlist(templated)),
                {:ok, sys_config} <- :erl_parse.parse_term(tokens),
                :ok <- validate_sys_config(sys_config),
-               merged <- Mix.Config.merge(base_config, sys_config) do
+               merged <- Config.Reader.merge(base_config, sys_config) do
```
Comment (CristhianRodriguezMolina) [inline lib/distillery/releases/assembler.ex:658]: Did you verify if these two `merge/2` functions work the same?

**Review by CristhianRodriguezMolina — CHANGES_REQUESTED** [review]

File: lib/distillery/releases/assembler.ex (hunk 2 above)
Comment (FelipeGuzmanSierra) [reply lib/distillery/releases/assembler.ex:658]: Hello @CristhianRodriguezMolina, yes, the module `Mix.Config` is deprecated, and the `Config.Reader` module from the elixir standard library is recommended to work with configuration files. In this case, both functions do the same thing. More about the modules can be found here:
- Mix.Config: https://hexdocs.pm/mix/1.12.3/Mix.Config.html#merge/2
- Config.Reader: https://hexdocs.pm/elixir/1.12/Config.Reader.html#merge/2

Comment (CristhianRodriguezMolina) [reply lib/distillery/releases/assembler.ex:658]: Understood, thank you.

**Review by CristhianRodriguezMolina — APPROVED** [review]

**Pull request merged**